src/test/java/org/scijava/module/ModuleServiceTest.java

```diff
@@ -158,6 +159,66 @@ public void testGetSingleInput() throws ModuleException {
 		assertSame(info.getInput("double2"), singleDouble);
 	}
 
+	@Test
+	public void testSaveAndLoad() {
+
+		List<Object[]> parameterizations = new ArrayList<>();
+		parameterizations.add(new Object[] { //
+			double[].class, new double[] {} //
+		});
+		parameterizations.add(new Object[] { //
+			double[].class, new double[] { 1., 2., 3. } //
+		});
+		parameterizations.add(new Object[] { //
+			Double[].class, new Double[] {} //
+		});
+		parameterizations.add(new Object[] { //
+			Double[].class, new Double[] { 1., 2., 3. } //
+		});
+
+		for (Object[] params : parameterizations) {
+			Class<?> type = (Class<?>) params[0];
+			Object expected = params[1];
+			// Get a ModuleItem of the right type
+			MutableModule m = new DefaultMutableModule();
+			m.getInfo().addInput(new DefaultMutableModuleItem<>(m, "a", type));
+			@SuppressWarnings("unchecked")
+			final ModuleItem<Object> item = (ModuleItem<Object>) moduleService
+				.getSingleInput(m, type);
+			// Save a value to the ModuleItem
+			moduleService.save(item, expected);
+			// Load that value from the ModuleItem
+			Object actual = moduleService.load(item);
+			// Assert equality
+			if (expected.getClass().isArray()) assertArrayEquality(expected, actual);
+			else assertEquals(expected, actual);
+		}
+	}
+
+	private void assertArrayEquality(Object arr1, Object arr2) {
+		// Ensure that both Objects are arrays of the same type!
+		assertEquals(arr1.getClass(), arr2.getClass());
+		assertTrue(arr1.getClass().isArray());
+
+		// We must check primitive arrays as they cannot be cast to Object[]
+		if (arr1 instanceof boolean[]) //
```

Comment:
Using `{ }` around the conditionals here would better match the style elsewhere in SciJava Common, e.g. in `ArrayUtils`.

Reply:
Fixed!@